File: Maximum Odd Binary Number.py
class Solution:
    def maximumOddBinaryNumber(self, s: str) -> str:
        n = len(s)
        count = 0
        for i in range(n):
            if s[i] == '1':
                count += 1

        num = []
        for i in range(n - 1):
            if count > 1:
                num.append("1")
                count -= 1
            else:
                num.append("0")

        num.append("1")
        return ''.join(num)

File: Maximum Odd Binary Number.cpp
class Solution {
public:
    string maximumOddBinaryNumber(string s) {
        int n = s.size();
        int count = 0;
        for (int i=0; i<n; i++) {
            if (s[i] == '1') count++;
        }

        string ns;
        ns.reserve(n);
        stringstream ss(ns);
        n--;
        while (n > 0) {
            if (count > 1) {
                ss << '1';
                count--;
            } else {
                ss << '0';
            }
            n--;
        }
        ss << '1';
        return ss.str();
    }
};
